deploy/charts/cert-manager/templates/_helpers.tpl

```diff
@@ -181,8 +181,26 @@ See https://github.com/cert-manager/cert-manager/issues/6329 for a list of linke
 */}}
 {{- define "image" -}}
 {{- $defaultTag := index . 1 -}}
+{{- $global := index . 2 -}}
 {{- with index . 0 -}}
-{{- if .registry -}}{{ printf "%s/%s" .registry .repository }}{{- else -}}{{- .repository -}}{{- end -}}
+{{- if .repository -}}
+    {{- if .registry -}}
+        {{- printf "%s/%s" .registry .repository -}}
```

Comment:
I feel like `{{- printf "%s/%s" .registry .name -}}` makes more sense here maybe? However, that might not b e backwards compatible. Is there something we can do to make this make sense?

Reply:
Do you think this is a correct use of the variables?
```
registry: quay.io/jetstack
repository: quay.io/jetstack/cert-manager-startupapicheck
name: cert-manager-startupapicheck
```

Reply:
I tried to think it through when I made this, I really didn't like the original construction and I couldn't think of a way to make it work with what I'm adding. 

I think the easiest thing is to make this a breaking version change. We could drop support entirely for the old stuff and rely on the helm validation to die to force people to reconfigure with the new value names. 

That said, I'm open to other ideas.

Reply:
The problem I have is that users can set `.registry` and `.repository` today, if they do, I can't actually change what happens, I'm basically pinned to that behavior. That's what this `if` preserves.

Reply:
I think that we might want to drop support for the `registry` field. It is not very useful in its current form.